addons/FlashCards/src/presenter.js

```diff
@@ -0,0 +1,468 @@
+        $(presenter.$view.find(".flashcards-card-audio-button-back")).addClass("disabled");
+        if(presenter.Cards[presenter.state.currentCard-1].AudioBack != ""){
+            $(presenter.$view.find(".flashcards-card-audio-wrapper-back")).show();
+            if (presenter.audioElementBack.canPlayType("audio/mpeg")) {
+                presenter.audioElementBack.setAttribute("src",presenter.Cards[presenter.state.currentCard-1].AudioBack);
+                presenter.audioElementBack.oncanplay = function() {
+                    $(presenter.$view.find(".flashcards-card-audio-button-back")).removeClass("disabled");
+                };
+            } 
+        }else{
+            presenter.audioElementBack.setAttribute("src","");
+            $(presenter.$view.find(".flashcards-card-audio-wrapper-back")).hide();
+        }
+    };
+
+    presenter.show = function(){
+        presenter.setVisibility(true);
+    };
+
+    presenter.hide = function(){
+        presenter.setVisibility(false);
+    };
+
+    presenter.setVisibility = function(isVisible) {
+        presenter.state.isVisible = isVisible;
+        presenter.$view.css("visibility", isVisible ? "visible" : "hidden");
+    };
+
+    presenter.updateVisibility = function() {
+        (presenter.state.isVisible) ? presenter.show() : presenter.hide();
+    };
+
+    presenter.setShowErrorsMode = function(){
+        presenter.removeClickHandlers();
+    }; 
+
+    presenter.setWorkMode = function(){
+        presenter.addClickHandlers();
+    };
+
+    presenter.ShowOnlyFavourites = function(){
+        presenter.state.ShowOnlyFavourites = true;
+        presenter.showCard(presenter.state.currentCard);
+    };
+    presenter.ShowAllCards = function(){
+        presenter.state.ShowOnlyFavourites = false;
+        presenter.showCard(presenter.state.currentCard);
+    };
+
+    presenter.resetFavourites = function(){
+        presenter.state.ShowOnlyFavourites = false;
+        $(presenter.Cards).each(function(k,v){
+            presenter.state.cardsFavourites[k] = false;
+        });
+        $(presenter.$view.find(".flashcards-button-favourite")).removeClass("flashcards-button-selected");
+    };
+    
+    presenter.reset = function() {
+        presenter.state.ShowOnlyFavourites = false;
+        presenter.state.currentCard = presenter.configuration.currentCard;
+
+        presenter.state.isVisible = presenter.configuration.isVisible;
+        presenter.updateVisibility();
+
+        $(presenter.model.Cards).each(function(k,v){
+            presenter.state.cardsScore[k] = 0;
+            presenter.state.cardsFavourites[k] = false;
+        });
+
+        presenter.showCard(presenter.state.currentCard);
+    };
+    
+    presenter.getErrorCount = function(){
+        var errors = null;
+        if(presenter.configuration.IsActivity){
+            errors = 0;
+            $(presenter.state.cardsScore).each(function(k,v){

```

Comment:
v nie jest wykorzystywane. Jeżeli aktualny element w each nie jest ci potrzebny, to ten argument funkcji może zostać pominięty.